### PR TITLE
JsonSchema: date and time

### PR DESCRIPTION
**Pull request by yufeih**
Upgrade test spec

https://dev.azure.com/ceapex/Engineering/_workitems/edit/107458/

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4914)

### CODE REVIEW

File: src/docfx/lib/schema/JsonSchemaValidator.cs [hunk 1]
```diff
@@ -214,10 +215,26 @@ private void ValidateString(JsonSchema schema, string name, JValue scalar, strin
...
+                    break;
+
                 case JsonSchemaStringFormat.DateTime:
-                    if (!DateTime.TryParse(str, out var _))
+                    if (!DateTime.TryParse(str, CultureInfo.InvariantCulture, DateTimeStyles.None, out var _))
                         errors.Add((name, Errors.FormatInvalid(JsonUtility.GetSourceInfo(scalar), str, JsonSchemaStringFormat.DateTime)));
                     break;
+
+                case JsonSchemaStringFormat.Date:
+                    if (!Regex.IsMatch(str, @"^(\d{4})-(\d{2})-(\d{2})$"))
```
Comment (mingwli) [inline src/docfx/lib/schema/JsonSchemaValidator.cs]: >(\d{2})-(\d{2}) [](start = 55, length = 15)

is this too direct.?

Comment (yufeih) [reply src/docfx/lib/schema/JsonSchemaValidator.cs]: See https://github.com/dotnet/docfx/pull/4914/files#diff-85968dd7fbc00463b422b61d04bdf8f8R13, `DateTime.Parse` is too permissive with respect to JSON schema spec.

Comment (mingwli) [reply src/docfx/lib/schema/JsonSchemaValidator.cs]: I mean for date like '1990-13-99'

File: src/docfx/lib/schema/JsonSchemaValidator.cs [hunk 2]
```diff
@@ -240,10 +241,26 @@ private void ValidateString(JsonSchema schema, string name, JValue scalar, strin
...
+                        errors.Add((name, Errors.FormatInvalid(JsonUtility.GetSourceInfo(scalar), str, JsonSchemaStringFormat.Date)));
+                    break;
+
+                case JsonSchemaStringFormat.Time:
+                    if (!DateTime.TryParse(str, CultureInfo.InvariantCulture, DateTimeStyles.NoCurrentDateDefault, out var time) || time.Date != default)
+                        errors.Add((name, Errors.FormatInvalid(JsonUtility.GetSourceInfo(scalar), str, JsonSchemaStringFormat.Time)));
+                    break;
+
+                default:
+                    throw new NotSupportedException($"JsonSchema: format '{schema.Format}'");
```
Comment (LowEntropyBody) [inline src/docfx/lib/schema/JsonSchemaValidator.cs]: Shall we throw exception or ignore?

Comment (yufeih) [reply src/docfx/lib/schema/JsonSchemaValidator.cs]: yep, we can just ignore it.

**Pull request merged**